pageserver/src/tenant.rs

```diff
@@ -730,6 +795,28 @@ impl Tenant {
                         make_broken(&tenant_clone, anyhow::anyhow!(e));
                     }
                 }
+                attach_timer.observe_duration();
+
+                // If we are doing an opportunistic warmup attachment at startup, initialize
+                // logical size at the same time.  This is better than starting a bunch of idle tenants
+                // with cold caches and then coming back later to initialize their logical sizes.
+                //
+                // It also prevents the warmup proccess competing with the concurrency limit on
+                // logical size calculations: if logical size calculation semaphore is saturated,
+                // then warmup will wait for that before proceeding to the next tenant.
+                if let AttachType::Warmup(_permit) = attach_type {
+                    let mut futs = FuturesUnordered::new();
```

Comment:
I have some unspecific FUD around FuturesUnordered. It seems like the right tool for the job, but, I'm not that familiar with the ecosystem. @koivunej

Reply:
I think the only spicy thing is if you drop it without completing all of them: in this context it's just running all the futures to completion, so should be pretty uncontroversial